Update XNNPACK Version

mcr229 opened the pull request:
Updating XNNPACK Version to 4ea82e595b36106653175dcb04b2aa532660d0d8

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @albanD

submodule update

mcr229 commented on cmake/Dependencies.cmake:
```diff
@@ -516,6 +516,9 @@ if(USE_XNNPACK AND NOT USE_SYSTEM_XNNPACK)
     # Disable I8MM For CI since clang 9 does not support neon i8mm.
     set(XNNPACK_ENABLE_ARM_I8MM OFF CACHE BOOL "")
 
+    # Disable avxvnni int8
```
this was breaking some CI so I just disabled. I forgot which one and why though :(

mcr229 commented on third_party/xnnpack_src_defs.bzl:
```diff
@@ -2,406 +2,1435 @@
 
-PROD_ARMSIMD32_MICROKERNEL_SRCS = [
-    "XNNPACK/src/amalgam/gen/armsimd32.c",
+XNNPACK_SRCS = [
```
yea I think I accidentally still ran the gen_wrappers script which edited these. These are no longer used even though they are modified. I will delete everything from that in a follow up diff.

huydhn commented:
I'm pretty sure that the https://github.com/pytorch/pytorch/actions/runs/11833440031/job/33005664656 is unrelated and it looks like an infra issue with the runner

mcr229 committed:
Update XNNPACK Version (#139913)

Summary:
Updating XNNPACK Version to 4ea82e595b36106653175dcb04b2aa532660d0d8

cc jgong5 mingfeima XiaobingSuper sanchitintel ashokei jingxu10 albanD

Pull Request resolved: https://github.com/pytorch/pytorch/pull/139913

Reviewed By: digantdesai, huydhn

Differential Revision: D65785169

Pulled By: mcr229

izaitsevfb commented:
@pytorchbot merge
